cyclonedx_py/parser/poetry.py

```diff
@@ -52,12 +55,22 @@ def __init__(
         debug_message('processing poetry_lock')
         for package in poetry_lock['package']:
             debug_message('processing package: {!r}', package)
+
+            if package['category'] == "dev":
```

Comment:
:x: do not use the literal string here.

So lets generate an Enum that holds well-known values for poetry "group".
and let `omit_category` be a list of values in that Enum.
The enum values are probably those mentioned in https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx/poetry.md